api/envoy/api/v2/route/route.proto

```diff
@@ -585,6 +585,16 @@ message RouteAction {
       // Connection properties hash policy.
       ConnectionProperties connection_properties = 3;
     }
+
+    // If true, and there is already a hash computed, ignore rest of the
+    // list of hash polices.
+    // For example, if the following hash methods are configured:
+    //   [(specifier, terminal),...]
+    //   [(Header A,    true),
+    //    (Cookie B,    false),
+    //    (Cookie C,    false)]
+    // The generateHash process ends after computing "header A", as it's a terminal policy.
```

Comment:
Can you add to the commit message and comments here the motivation for the terminal attribute; is there some real-world use case where this is important? Knowing what you are working on, I'm sure there is, but it would be helpful to share.